src/lib/loaders/api/loader_without_authentication_factory.js

```diff
@@ -41,7 +41,7 @@ export const apiLoaderWithoutAuthenticationFactory = (
             const clock = timer(key)
             clock.start()
             return new Promise((resolve, reject) => {
-              cache.get(key).then(
+              return cache.get(key).then(
```

Comment:
It's weird that the rule requires a return because promises are pretty nice for managing async flow independent of data being passed back up. I wonder what the reasoning is?

Reply:
Ah https://github.com/xjamundx/eslint-plugin-promise#rules

Reply:
We had surprisingly few issues with most of those rules, so go us! The nested promises and callbacks-in-promises are fair warnings that, if we fix them, should make the code a bit more reasonable.